lib/fabrik/VirtualHostInstance.js

```diff
@@ -0,0 +1,102 @@
+const _ = require('lodash');
+const Promise = require('bluebird');
+const BaseInstance = require('./BaseInstance');
+const bosh = require('../bosh');
+const utils = require('../utils');
+const mapper = require('./VirtualHostRelationMapper');
+const catalog = require('../models').catalog;
+const logger = require('../logger');
+const errors = require('../errors');
+const NotFound = errors.NotFound;
+
+class VirtualHostInstance extends BaseInstance {
+  constructor(guid, manager) {
+    super(guid, manager);
+    this.director = bosh.director;
+    this.mapper = mapper.VirtualHostRelationMapper;
+  }
+
+  get async() {
+    return false;
+  }
+
+  initialize() {
+    return this.getDeploymentNameForInstanceId(this.guid)
+      .then(deploymentName => this.deploymentName = deploymentName);
+  }
+
+  create(params) {
+    const rabbitmqInstanceName = params.parameters.dedicated_rabbitmq_instance;
+    return this.cloudController.getServiceInstances({
```

Comment:
can you move this piece of code between Line 30 - 40, into cloud controller client into a new method named getInstanceByName(name, space_guid)? The rules of forming query & syntax to query CF should be within cloud controller client class and not spill outside of it.  This method if successful should return  `serviceInstances[0]` else `throw new NotFound(``Service instance with name ${instanceName} not found``);`